packages/taquito-wallet-connect-2/src/taquito-wallet-connect-2.ts

```diff
@@ -0,0 +1,690 @@
+      topic: session.topic,
+      chainId: `${TEZOS_PLACEHOLDER}:${network}`,
+      request: {
+        method: PermissionScopeMethods.TEZOS_SIGN,
+        params: {
+          expression,
+          signingType: SigningType.RAW,
+        },
+      },
+    });
+    return signature;
+  }
+
+  /**
+   * @description Once the session is establish, send payload to be approved and signed by the wallet.
+   * @error MissingRequiredScope is thrown if permission to sign payload was not granted
+   */
+  async signPayload(params: {
+    signingType?: SigningType;
+    payload: string;
+    sourceAddress?: string;
+  }) {
+    const session = this.getSession();
+    if (!this.getPermittedMethods().includes(PermissionScopeMethods.TEZOS_SIGN)) {
+      throw new MissingRequiredScope(PermissionScopeMethods.TEZOS_SIGN);
+    }
+    const network = this.getActiveNetwork();
+    const account = await this.getPKH();
+    this.validateNetworkAndAccount(network, account);
+    const signature = await this.signClient.request<string>({
+      topic: session.topic,
+      chainId: `${TEZOS_PLACEHOLDER}:${network}`,
+      request: {
+        method: PermissionScopeMethods.TEZOS_SIGN,
+        params: {
+          account: params.sourceAddress ?? account,
+          expression: params.payload,
+          signingType: params.signingType,
+        },
+      },
+    });
+    console.log(signature);
+    return signature;
+  }
+
+  /**
+   * @description Return all connected accounts from the active session
+   * @error NotConnected if no active session
+   */
+  getAccounts() {
+    return this.getTezosNamespace().accounts.map((account) => account.split(':')[2]);
+  }
+
+  /**
+   * @description Set the active account.
+   * Must be called if there are multiple accounts in the session and every time the active account is switched
+   * @param pkh public key hash of the selected account
+   * @error InvalidAccount thrown if the pkh is not part of the active accounts in the session
+   */
+  setActiveAccount(pkh: string) {
+    if (!this.getAccounts().includes(pkh)) {
+      throw new InvalidAccount(pkh);
+    }
+    this.activeAccount = pkh;
+  }
+
+  /**
+   * @description Access the public key hash of the active account
+   * @error ActiveAccountUnspecified thrown when there are multiple Tezos account in the session and none is set as the active one
+   */
+  async getPKH() {
+    if (!this.activeAccount) {
+      this.getSession();
+      throw new ActiveAccountUnspecified();
+    }
+    return this.activeAccount;
+  }
+
+  // TODO need test-dapp test
+  /**
+   * @description Access the public key of the active account
+   * @error ActiveAccountUnspecified thrown when there are multiple Tezos account in the session and none is set as the active one
+   */
+  async getPK() {
+    if (!this.activeAccount) {
+      this.getSession();
+      throw new ActiveAccountUnspecified();
+    }
+    return this.getSession().self.publicKey;
```

Comment:
@hui-an-yang 
When looking at the session returned by Kukai `session.self.publicKey` is not the Tezos public key of the active account:
![image](https://github.com/user-attachments/assets/1c731aa7-dd83-4798-9794-d1ae15ba9c8e)